controllers/awsmachine_controller.go

```diff
@@ -553,16 +557,18 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	if machineScope.InstanceIsOperational() {
 		machineScope.SetAddresses(instance.Addresses)
 
-		existingSecurityGroups, err := ec2svc.GetInstanceSecurityGroups(*machineScope.GetInstanceID())
-		if err != nil {
-			return ctrl.Result{}, err
-		}
+		if !machineScope.IsExternalInfraCluster() {
```

Comment:
Do you think we will need to do something similar for machine pools? If so, then we probably want to move the [getInfraCluster](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f02e98508e6b09b0c5aaebcf631469fa603c9695/controllers/awsmachine_controller.go#L770) to somewhere where it can be shared.